### ISSUE TITLE
Out-of-Memory errors with certain inputs since 14.2.0

### ISSUE THREAD

**Issue by Systemcluster** [spencermountain/compromise#926]
Certain inputs with certain terms analyses result in out-of-memory errors since `14.2.0`.

Minimal reproduction case:
```js
import compromise from "compromise/three"

const string = `The hours have passed like stones being pushed up a mountain. For all of the luxury that surrounds us, I can't shake this feeling of unease that's slowly creeping in through the back of my mind. I can tell that Johna and Temmy have noticed it as well—it's just something about the air here that makes me uneasy. Joanna feigns disinterest but behind her shades she's studying the surroundings like the seasoned detective she is.`

const text = (string + '\n').repeat(50)

const usedWords = new Map()
const data = compromise(text)
for (const term of data
	.terms()
	.not('#Pronoun')
	.not('#Preposition')
	.not('#Conjunction')
	.not('#Determiner')
	.json()) {
	const word = term.terms[0].text
	if (word !== '') usedWords.set(word, (usedWords.get(word) ?? 0) + 1)
}	
console.log(usedWords)	
```

With `14.1.1`, it finishes in under a second. With `14.2.0`, it takes around a minute until it crashes with OOM.

```
<--- Last few GCs --->

[34036:000001DB6B78BEC0]    52131 ms: Mark-sweep 4049.1 (4142.6) -> 4047.1 (4142.9) MB, 1528.1 / 0.0 ms  (average mu = 0.286, current mu = 0.153) allocation failure; scavenge might not succeed
[34036:000001DB6B78BEC0]    53419 ms: Mark-sweep 4050.1 (4143.6) -> 4048.0 (4143.9) MB, 1285.3 / 0.0 ms  (average mu = 0.180, current mu = 0.002) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF6FEC594EF node_api_throw_syntax_error+174703
 2: 00007FF6FEBE79C6 v8::internal::wasm::WasmCode::safepoint_table_offset+67350
 3: 00007FF6FEBE889D node::OnFatalError+301
 4: 00007FF6FF679B6E v8::Isolate::ReportExternalAllocationLimitReached+94
 5: 00007FF6FF664EF2 v8::Isolate::Exit+674
 6: 00007FF6FF4E6D9C v8::internal::EmbedderStackStateScope::ExplicitScopeForTesting+124
 7: 00007FF6FF4F407D v8::internal::Heap::PublishPendingAllocations+1117
 8: 00007FF6FF4F1107 v8::internal::Heap::PageFlagsAreConsistent+3367
 9: 00007FF6FF4E3837 v8::internal::Heap::CollectGarbage+2039
10: 00007FF6FF4FA1F3 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath+2099
11: 00007FF6FF4FAA9D v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath+93
12: 00007FF6FF50A2D3 v8::internal::Factory::NewFillerObject+851
13: 00007FF6FF1FBB9F v8::internal::DateCache::Weekday+1071
14: 00007FF6FF717211 v8::internal::SetupIsolateDelegate::SetupHeap+558193
15: 00007FF67F9CDD84```

**Comment by spencermountain** [comment]
whoa!
sorry. will take a look at this today.

**Comment by spencermountain** [comment]
this should be fixed now in `14.2.1`, let me know if anything else comes up
cheers
